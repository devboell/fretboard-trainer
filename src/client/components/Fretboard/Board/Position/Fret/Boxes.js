import { css } from 'styled-components'
import SkinWrapper from './SkinWrapper'

const Boxes = SkinWrapper.extend`
  height: 80%;
  width: 90%;
  margin: 10% 5%;
  border: 1px solid darkgray;
  border-radius: 3px;
  ${props => props.isHighlighted && css`background-color: ${props.theme.highlight};`}
`

export default Boxes
