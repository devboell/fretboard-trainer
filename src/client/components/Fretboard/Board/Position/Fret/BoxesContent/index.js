import React from 'react'
import pt from 'prop-types'
import Wrapper from './Wrapper'


const BoxesContent = ({ note }) =>
  <Wrapper>
    {note}
  </Wrapper>

BoxesContent.propTypes = {
  note: pt.string.isRequired,
}

export default BoxesContent
