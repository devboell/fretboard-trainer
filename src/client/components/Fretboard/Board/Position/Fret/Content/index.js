import React from 'react'
import pt from 'prop-types'
import Wrapper from './Wrapper'


const Content = ({ note }) =>
  <Wrapper>
    {note}
  </Wrapper>

Content.propTypes = {
  note: pt.string.isRequired,
}

export default Content
