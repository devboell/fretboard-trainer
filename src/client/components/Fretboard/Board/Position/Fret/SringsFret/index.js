import React from 'react'
import pt from 'prop-types'
import Wrapper from './Wrapper'


const StringsFret = ({ note }) =>
  <Wrapper>
    {note}
  </Wrapper>

StringsFret.propTypes = {
  note: pt.string.isRequired,
}

export default StringsFret
