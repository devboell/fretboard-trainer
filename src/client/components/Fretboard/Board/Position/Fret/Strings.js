import { css } from 'styled-components'
import SkinWrapper from './SkinWrapper'

const highlightCSS = ({ theme, status }) => {
  const radiusPerc = 80
  const radius = Math.floor((theme.stringHeight / 100) * radiusPerc)

  return css`
    width: ${radius}px;
    height: ${radius}px;
    border-radius: 50%;
    background-color: ${theme.highlight[status]};
  `
}

const defaultCSS = ({ theme }) =>
  css`
    padding: 0 5%;
    background-color: ${theme.background};
  `

const Strings = SkinWrapper.extend`
  ${props => (props.isHighlighted
    ? highlightCSS(props)
    : defaultCSS(props))} 
  `

export default Strings
