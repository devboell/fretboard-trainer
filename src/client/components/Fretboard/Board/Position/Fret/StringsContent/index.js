import React from 'react'
import pt from 'prop-types'
import Wrapper from './Wrapper'


const StringsContent = ({ note }) =>
  <Wrapper>
    {note}
  </Wrapper>

StringsContent.propTypes = {
  note: pt.string.isRequired,
}

export default StringsContent
