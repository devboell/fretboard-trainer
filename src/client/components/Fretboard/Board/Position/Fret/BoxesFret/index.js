import React from 'react'
import pt from 'prop-types'
import BoxesContent from '../BoxesContent'
import Wrapper from './Wrapper'


const BoxesFret = ({ note, showNotes, isHighlighted }) =>
  <Wrapper isHighlighted={isHighlighted}>
    {showNotes &&
      <BoxesContent note={note} />
    }
  </Wrapper>

BoxesFret.propTypes = {
  note: pt.string.isRequired,
  showNotes: pt.bool.isRequired,
  isHighlighted: pt.bool.isRequired,
}

export default BoxesFret
