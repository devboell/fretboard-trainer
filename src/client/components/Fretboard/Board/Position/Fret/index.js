import React from 'react'
import pt from 'prop-types'
import BoxesFret from './BoxesFret'
import StringsFret from './StringsFret'


import Wrapper from './Wrapper'

/* eslint-disable react/prefer-stateless-function */
class Fret extends React.Component {
  render() {
    const { type, showNotes, isHighlighted } = this.context
    const Outer = type === 'boxes' ? BoxesFret : StringsFret

    return (
      <Wrapper>
        <Outer
          note={this.props.note}
          showNotes={showNotes}
          isHighlighted={isHighlighted}
        />
      </Wrapper>)
  }
}


Fret.propTypes = {
  note: pt.string.isRequired,
}

Fret.contextTypes = {
  type: pt.string.isRequired,
  showNotes: pt.bool.isRequired,
  isHighlighted: pt.bool.isRequired,
}

export default Fret
