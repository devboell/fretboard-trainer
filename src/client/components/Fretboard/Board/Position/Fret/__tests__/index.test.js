import React from 'react'

import Fret from '../index'


describe('Fret component', () => {
  const props = {
    note: 'E2',
  }
  const defaultContext = {
    selectedNotes: ['C4'],
    type: 'boxes',
    showNotes: true,
    showSelection: true,
  }

  it('defaultContext, snapshot', () => {
    const wrapper = shallow(<Fret {...props} />, { context: defaultContext })
    expect(wrapper).toMatchSnapshot()
  })

  it('type=strings, showNotes-false, snapshot', () => {
    const context = { ...defaultContext, type: 'strings', showNotes: false }
    const wrapper = shallow(<Fret {...props} />, { context })
    expect(wrapper).toMatchSnapshot()
  })

  it('showSelection=false, note is selected, snapshot', () => {
    const context = {
      ...defaultContext,
      showSelection: false,
      selectedNotes: ['E2'],
    }
    const wrapper = shallow(<Fret {...props} />, { context })
    expect(wrapper).toMatchSnapshot()
  })

  it('note is selected, snapshot', () => {
    const context = {
      ...defaultContext,
      selectedNotes: ['E2'],
    }
    const wrapper = shallow(<Fret {...props} />, { context })
    expect(wrapper).toMatchSnapshot()
  })
})
