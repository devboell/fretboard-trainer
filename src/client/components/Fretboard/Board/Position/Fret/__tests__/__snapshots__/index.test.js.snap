// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fret component defaultContext, snapshot 1`] = `
<styled.div>
  <styled.div
    isHighlighted={true}
    status="oct2"
  >
    <Content
      content="E2"
    />
  </styled.div>
</styled.div>
`;

exports[`Fret component note is selected, showOctaves=false, snapshot 1`] = `
<styled.div>
  <styled.div
    isHighlighted={true}
    status="selected"
  >
    <Content
      content="E2"
    />
  </styled.div>
</styled.div>
`;

exports[`Fret component showSelection=false, note is selected, snapshot 1`] = `
<styled.div>
  <styled.div
    isHighlighted={true}
    status="selected"
  >
    <Content
      content="E2"
    />
  </styled.div>
</styled.div>
`;

exports[`Fret component type=strings, showNotes-false, snapshot 1`] = `
<styled.div>
  <styled.div
    isHighlighted={true}
    status="oct2"
  />
</styled.div>
`;
